Listeners/ExceptionListener.php

```diff
@@ -0,0 +1,102 @@
+<?php
+namespace Infrastructure\Listeners;
+
+use Infrastructure\Exceptions\HttpExceptionInterface;
+use Infrastructure\Models\HttpExceptionFactory;
+use Infrastructure\Services\ErrorHandlerInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class ExceptionListener implements EventSubscriberInterface
+{
+    protected $logger;
+    protected $debug;
+    protected $errorHandler;
+
+    /**
+     * ExceptionListener constructor.
+     * @param $errorHandler
+     * @param LoggerInterface|null $logger
+     * @param bool $debug
+     */
+    public function __construct(ErrorHandlerInterface $errorHandler, LoggerInterface $logger = null, $debug = false)
+    {
+        $this->logger = $logger;
+        $this->debug = $debug;
+        $this->errorHandler = $errorHandler;
+    }
+
+    /**
+     * @return array
+     */
+    public static function getSubscribedEvents()
+    {
+        return array(
+            KernelEvents::EXCEPTION => array(
+                array('logKernelException', 0),
```

Comment:
Correct me if I'm wrong. It means that a log first onKernel second?

Reply:
You are right. 
It's long discussion what we priorities here. Some one think that logs is more important to track and than improve code.
No logger is empty.